pkg/ibm/ibmp_helpers_test.go

```diff
@@ -9,35 +9,115 @@ import (
 )
 
 var _ = Describe("IBM Power Cloud Helper Functions", func() {
-	var instance *models.PVMInstance
 
-	BeforeEach(func() {
-		instance = &models.PVMInstance{
-			PvmInstanceID: ptr("id"),
-		}
+	Describe("doesInstanceHaveTaskRun helper function tests", func() {
+		var instance *models.PVMInstance
+
+		BeforeEach(func() {
+			instance = &models.PVMInstance{
+				PvmInstanceID: ptr("id"),
+			}
+		})
+		DescribeTable("Determine if a VM instance is linked to a non-existing TaskRun",
+			func(log logr.Logger, userTags models.Tags, existingTaskRuns map[string][]string, expectedResult bool) {
+				ibmp := IBMPowerDynamicConfig{}
+				instance.UserTags = userTags
+				Expect(ibmp.doesInstanceHaveTaskRun(log, instance, existingTaskRuns)).Should(Equal(expectedResult))
```

Comment:
minor: sounds like  we can move `logr.Discard()` right here and make 1 param entry less ?